### PR TITLE
Android: Dual Compile for Qt6 and Qt5 

### PR DESCRIPTION
**Pull request by strseb**
This is a replicate of #2387 - As CI was failing- since it was a pr from a fork. So pulled it in this branch :)

### CODE REVIEW

File: .github/workflows/android_qt6.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  compile-android-qt6:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false # Don't cancel other jobs
+      matrix:
+        config:
+            #- { name: Release, args: "", apk: .tmp/src/android-build/build/outputs/apk/release/, sym: true, sign: true  }
+            - { name: Debug, args: "--debug", apk: .tmp/src/android-build/mozillavpn.apk, sym: false, sign: false}
+        arch: 
+          # Qt6 names their arch diffrent then what qmake expects ...
```
Comment (bakulf) [inline .github/workflows/android_qt6.yaml:23]: Different/differently? And "e" is missing

File: .github/workflows/android_qt6.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          - {name: "armv7", qmake: "armeabi-v7a"}
+          - {name: "arm64_v8a", qmake: "arm64-v8a"}
+          - {name: "x86" , qmake: "x86"}
+          - {name: "x86_64", qmake: "x86_64"}
+        version: ["6.2.1"]
+    container:
+      image: ghcr.io/strseb/vpn-builder-android:android_${{matrix.arch.name}}_${{matrix.version}}
```
Comment (bakulf) [inline .github/workflows/android_qt6.yaml:30]: Can we move this under Mozilla?

File: .gitignore [hunk 3]
```diff
@@ -42,7 +43,7 @@ android/.gradle/
 android/.idea/
 android/build
 android/local.properties
-android/*/.cxx/
+android/.cxx/
```
Comment (bakulf) [inline .gitignore:46]: Is this right?

File: android/tunnel/CMakeLists.txt [hunk 4]
```diff
@@ -15,16 +15,16 @@ if(CCACHE_FOUND)
 endif(CCACHE_FOUND)
 
 # Work around https://github.com/android-ndk/ndk/issues/602
-set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fuse-ld=gold")
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fuse-ld=lld")
```
Comment (bakulf) [inline android/tunnel/CMakeLists.txt:18]: Why a different linker?

File: scripts/android_package.sh [hunk 5]
```diff
@@ -173,12 +180,14 @@ else
...
     CONFIG-=release \
     CONFIG+=qml_debug \
+    ANDROID_ABIS=$ARCH \
     $ADJUST \
     ..//mozillavpn.pro || die "Qmake failed"
 fi
 
 print Y "Compiling apk_install_target in .tmp/"
-make -j $JOBS sub-src-apk_install_target || die "Compile of QT project failed"
+make -s -j $JOBS sub-src-apk_install_target  || die "Compile of QT project failed"
```
Comment (bakulf) [inline scripts/android_package.sh:189]: Have you just add a space?

**Review by bakulf — APPROVED** [review]

File: .github/workflows/android_qt6.yaml (hunk 2 above)
Comment (strseb) [reply .github/workflows/android_qt6.yaml:30]: I am currently working on moving this part to taskcluster. This includes that we have the images in-tree :) - So this is just an intermediate state

File: android/tunnel/CMakeLists.txt (hunk 4 above)
Comment (strseb) [reply android/tunnel/CMakeLists.txt:18]: gold is depricated and was removed in the new ndk version we're using now

**Pull request merged**